packages/lesmis/src/components/DashboardReportModal.js

```diff
@@ -91,30 +92,31 @@ export const DashboardReportModal = ({
     dashboardGroupId,
     reportId,
     periodGranularity,
-    startDate: params.get('startDate'),
-    endDate: params.get('endDate'),
+    startDate,
+    endDate,
   });
 
-  useEffect(() => {
+  const handleDatesChange = (newStartDate, newEndDate) => {
     setParams({
-      startDate: null,
-      endDate: null,
-    });
-  }, [open]);
-
-  const handleDatesChange = (startDate, endDate) => {
-    setParams({
-      startDate,
-      endDate,
+      startDate: newStartDate,
+      endDate: newEndDate,
     });
   };
 
   const handleClickOpen = () => {
     setOpen(true);
```

Comment:
That's a good point. I tried refactoring the open state to be purely based on the presence of the reportId but then realised that it created a lag in the modal opening because the modal waits for the url to be set and then parent components to be re-rendered that use url location params before opening. I then went down a huge rabbit hole trying to optimise the render flow from urls with minimal success. Overall I've left the `open` in state.